### PR TITLE
Modify sync-in to also gather function descriptions

### PR DESCRIPTION
**Pull request by bethanyaconnor**
First PR in the process of translating function descriptions.

Process I'm following:
1. (This PR) Update the `sync-in` script to gather the function descriptions. I'm changing the structure very slightly so that it's structured like:
```
level_url: {
  function_definitions: {
    function_name: {
      name: "function_name"
      description: "function description"
    }
  }
 }
```
Note that the key changed from `function_names` to `function_definitions`. I did this to make the transition a bit easier, plus it makes more sense.
2. Before running a sync that contains this PR, upload a file with the existing function names. This will allow duplication to work correctly.
3. Run a sync with this change. The out step will produce `function_defintions.*.yml` and will not update `function_names.*.yml`. Check in the `function_defintions.*.yml` files.
4. Modify [localized_function_blocks](https://github.com/code-dot-org/code-dot-org/blob/783cca45fa6cd15b6ddc30de2095482507de473a/dashboard/app/models/levels/blockly.rb#L494) to read from the new translation files. As part of that PR, delete `function_names.*.yml`.

This process limits the management of PR/sync timing to step 2. The sync out needs to happened before step 4, but there isn't any complicated PR juggling (i.e. making sure the updated rendering logic the sync go out in the same DTP).

## Links

- [spec]()
- [jira]()

## Testing story

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: bin/i18n/sync-in.rb [hunk 1]
```diff
@@ -75,10 +75,13 @@ def get_i18n_strings(level)
...
       functions = blocks.xpath("//block[@type=\"procedures_defnoreturn\"]")
-      i18n_strings['function_names'] = Hash.new unless functions.empty?
+      i18n_strings['function_defintions'] = Hash.new unless functions.empty?
       functions.each do |function|
         name = function.at_xpath('./title[@name="NAME"]')
-        i18n_strings['function_names'][name.content] = name.content if name
+        description = function.at_xpath('./mutation/description')
+        i18n_strings['function_defintions'][name.content] = Hash.new
+        i18n_strings['function_defintions'][name.content]["name"] = name.content if name
+        i18n_strings['function_defintions'][name.content]["description"] = description.content if description
```
Comment (Hamms) [inline bin/i18n/sync-in.rb:84]: is it possible for there to be functions defined in two places that share the same name but have different descriptions?

Comment (bethanyaconnor) [reply bin/i18n/sync-in.rb:84]: In the same level? That seems unlikely but I haven't checked

Comment (Hamms) [reply bin/i18n/sync-in.rb:84]: oh, right! I forgot we broke these up by level

**Review by Hamms — APPROVED** [review]
Both code and process look good to me! I love having the process laid out here so explicitly, too!

**Comment by bethanyaconnor** [comment]
Uploaded a JSON with the function names to Crowdin -- all duplicates as expected. I'm going to merge this PR

**Pull request merged**